### PR TITLE
Do not cast sizes to int in Trashbin class

### PR DESCRIPTION
**Pull request by come-nc**
This helps with 32bits support

It should fix failing CI: https://github.com/nextcloud/server/actions/workflows/phpunit-32bits.yml

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by AndyScherzinger — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

File: apps/files_trashbin/lib/Trashbin.php [hunk 1]
```diff
@@ -306,8 +305,8 @@
 		}
 
 		$config = \OC::$server->getConfig();
-		$systemTrashbinSize = (int)$config->getAppValue('files_trashbin', 'trashbin_size', '-1');
-		$userTrashbinSize = (int)$config->getUserValue($owner, 'files_trashbin', 'trashbin_size', '-1');
+		$systemTrashbinSize = \OCP\Util::numericToNumber($config->getAppValue('files_trashbin', 'trashbin_size', '-1'));
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/files_trashbin/lib/Trashbin.php:308]: @come-nc shall still cast to int?

[the remark this answers is not in the document]
Comment (come-nc) [reply apps/files_trashbin/lib/Trashbin.php]: No, the point of `numericToNumber` is to cast to int|float, to support 32bit systems.
I pushed a refactor to please psalm (one needs to check that the config value is numeric before calling numericToNumber)

**Comment by skjnldsv** [comment]
https://github.com/nextcloud/server/pull/38717

**Comment by go2sh** [comment]
Are ypu really Sure to allow filesizes to be float? It can only represent 24bit filesizes accurately on 32 bit systems. Every filesize bigger will ge an error, which gets bigger with bigger filesizes. On can argue that for the trashbin size this might be OK, but for concrete files I doubt it.

**Comment by come-nc** [comment]
> Are ypu really Sure to allow filesizes to be float? It can only represent 24bit filesizes accurately on 32 bit systems. Every filesize bigger will ge an error, which gets bigger with bigger filesizes. On can argue that for the trashbin size this might be OK, but for concrete files I doubt it.

This is not a choice, this is how PHP works on 32bits. Any int bigger than what an int can contain is casted into a float. This causes type errors as soon as a file is bigger than 4G, or in this case when empty space in the Trashbin is bigger than 4G, which happens quite often.

**Pull request merged**